### PR TITLE
Python3 compatible

### PR DESCRIPTION
**Pull request by zlangley**

### CODE REVIEW

File: clrenv/lazy_env.py [hunk 1]
```diff
@@ -42,26 +44,27 @@ def __getitem__(self, key):
...
-        mode = filter(lambda m: m.find('=') == -1, mode)
+        assignments = tuple(m for m in mode if m.find('=') != -1)
+        mode = tuple(m for m in mode if m.find('=') == -1)
 
         overrides = []
         for a in assignments:
             overrides.append(a.split('=', 1))
 
-        allenvs = set(chain(*(x.keys() for x in y)))
+        allenvs = set(chain(*(list(x.keys()) for x in y)))
```
Comment (snarfed) [inline clrenv/lazy_env.py:60]: curious why the new `list(...)`s here and below are necessary. does the code break without them?

**Review by snarfed — APPROVED** [review]
❤️❤️❤️ python 3! woo! so exciting. btw beware, @youssefm might just kiss you if you actually get us all the way there. :P

File: clrenv/lazy_env.py (hunk 1 above)
Comment (zlangley) [reply clrenv/lazy_env.py:60]: They probably aren't necessary here, but `x.keys()` in Python 3 doesn't return a `list`. In general, this type of conversion prevents bugs from code that depend on `keys` returning a list.

Comment (snarfed) [reply clrenv/lazy_env.py:60]: understood, but these usages didn't look like they required lists. definitely just curious, not a big deal. no need to change now!

Comment (zlangley) [reply clrenv/lazy_env.py:60]: Yea, these changes are autogenerated by a script that is trying to preserve behavior in all cases. Once everything works under Python 3 we can revisit and remove unnecessary type conversions.

Comment (snarfed) [reply clrenv/lazy_env.py:60]: ahh makes sense now! if this is from e.g. `2to3`, i don't care about little things like `list(...)` then. i for one welcome our new robot overlords, and all that.

**Pull request merged**